src/Main.elm

```diff
@@ -1,33 +1,104 @@
 module Main exposing (..)
 
 import Browser
-import Html exposing (Html, text, div, h1, img)
-import Html.Attributes exposing (src)
+import Html exposing (..)
+import Html.Attributes exposing (class, href)
+import Html.Events exposing (..)
+import Http
+import Json.Decode as Decode exposing (Decoder, andThen, int, list, string, succeed, field)
+import Json.Decode.Pipeline exposing (required)
+import RemoteData exposing (RemoteData, WebData)
+
 
 
 ---- MODEL ----
 
 
+type alias Ids =
+    { id : Int }
+
+type alias Story =
+    { by : String
+    , descendants : Int
+    , id : Int
+    , kids : List Int
+    , score : Int
+    , time : Int
+    , title : String
+    , url : String
+    }
+
 type alias Model =
-    {}
+    { hackerNews : WebData (List Story)
+     , storyIds : WebData ( List Ids )
+     }
+
+
+emptyModel : Model
+emptyModel =
+    { hackerNews = RemoteData.Loading,
+     storyIds = RemoteData.Loading }
```

Comment:
You probably don't need to store it in model, because once we get IDs list, we'll use it to fetch the stories.